dbms/src/Common/FmtUtils.h

```diff
@@ -14,35 +14,43 @@
 
 #pragma once
 
+#include <fmt/compile.h>
 #include <fmt/format.h>
 
 namespace DB
 {
+
 class FmtBuffer
 {
 public:
     FmtBuffer() = default;
 
-    template <typename... Args>
-    FmtBuffer & fmtAppend(std::string_view fmt, Args &&... args)
+    template <typename FmtStr, typename... Args>
+    inline FmtBuffer & fmtAppend(FmtStr fmt, Args &&... args)
     {
-        fmt::vformat_to(std::back_inserter(buffer), fmt, fmt::make_format_args(std::forward<Args>(args)...));
+        fmt::format_to(std::back_inserter(buffer), fmt, std::forward<Args>(args)...);
```

Comment:
Have you compiled this code under c++20? I'm not sure if `format_to` works well here.